File: minimum-operations-to-make-the-array-increasing.ts
{
    const minOperations = function(nums) {
        let result = 0;

        for(let i = 0; i < nums.length - 1; i++) {
            if(nums[i + 1] <= nums[i]) {
                const diff = nums[i] - nums[i + 1] + 1;

                nums[i + 1] = nums[i + 1] +  diff;

                result += diff;
            }

        }

        return result
    };

    console.log(minOperations([1,1,1]))
}

File: minimum-operations-to-make-the-array-increasing.rs
pub fn min_operations(mut nums: Vec<i32>) -> i32 {
    let mut с = 0;

    for i in 0..nums.len(){
        if i != 0{
            if nums[i] <= nums[i-1]{                 
                с+= nums[i-1] - nums[i] + 1;
                nums[i] = nums[i-1] + 1;
            }
        }
    }

    с
}
